docs/multiple-namespaces.md

```diff
@@ -0,0 +1,115 @@
+# Multiple Namespace Support
+
+#### Motivation
+Kubernetes [Namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/)
+make it possible for a Kubernetes cluster to be partitioned and allocated to
+sub-groups of a larger team. These sub-teams can then deploy and manage
+infrastructure with finer controls of resources, security, configuration etc.
+Kubernetes allows for one or more ingress resources to be defined independently
+within each namespace.
+
+As of version 0.7 [Azure Application Gateway Kubernetes
+IngressController](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/README.md)
+(AGIC) can ingest events from and observe multiple namespaces. Should the AKS
+administrator decide to use [App
+Gateway](https://azure.microsoft.com/en-us/services/application-gateway/) as an
+ingress, all namespaces will use the same instance of App Gateway. A single
+installation of Ingress Controller will monitor accessible namespaces and will
+configure the App Gateway it is associated with.
+
+#### Defaults
+As of version 0.7, AGIC is configured to observe ingress resources from the
+*default* namespace or the namespace explicitly defined in the Helm
+configuration.
+
+#### Enable multiple namespace support
+To enable miltiple namespace support - modify the [helm-config.yaml](examples/helm-config.yaml) file in one of the following ways:
+  1. delete the `watchNamespace` key entirely from [helm-config.yaml](examples/helm-config.yaml) - AGIC will observe all namespaces
+  1. set `watchNamespace` to an empty string - AGIC will observe all namespaces
+  2. add multiple namespaces separated by a comma (`watchNamespace: default,secondNamespace`) - AGIC will observe these namespaces exclusively
+
+Install Helm template changes with: `helm install -f helm-config.yaml application-gateway-kubernetes-ingress/ingress-azure`
+
+Once deployed with the ability to observe multiple namespaces, AGIC will:
+  - list ingress resources from all accessible namespaces
+  - filter to ingress resources annotated with `kubernetes.io/ingress.class: azure/application-gateway`
+  - compose combined [App Gateway config](https://github.com/Azure/azure-sdk-for-go/blob/37f3f4162dfce955ef5225ead57216cf8c1b2c70/services/network/mgmt/2016-06-01/network/models.go#L1710-L1744)
+  - apply the config to the associated App Gateway via [ARM](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview)
+
+#### Conflicting Configurations
```

Comment:
As we are mentioning the conflicting scenario, should we mention what two ingresses can share without having conflicts ?

Reply:
Rewrote pieces of that to add clarity.